src/storage/src/sink/kafka.rs

```diff
@@ -902,14 +913,25 @@ impl KafkaSinkState {
         progress_emitted
     }
 
-    async fn update_status(&self, status: SinkStatus) {
-        update_status(&self.healthchecker, status).await;
+    fn update_status(&self, status: SinkStatus) {
+        if SinkStatus::can_transition(self.latest_status.as_ref(), &status) {
```

Comment:
why was the call for `can_transition` moved into the `sink` impl? Ideally it wouldn't need to remember to call this, instead it would just emit updates. To me it seems like the `healthcheck` module should transition to being a wrapper over `object_status_updates`, which can filter for transitions that don't need to send a message to the controller

Reply:
Yeah, I see your point. Happy to update there.